kevm-pyk/src/kevm_pyk/kdist/_kdist.py

```diff
@@ -50,118 +52,158 @@ def _dist_dir() -> Path:
     return xdg_cache_home() / f'kdist-{digest}'
 
 
-def _plugins() -> dict[str, dict[str, Target]]:
-    import importlib
-    from importlib.metadata import entry_points
+class TargetId(NamedTuple):
+    plugin: str
+    target: str
 
-    plugins = entry_points(group='kdist')
+    @staticmethod
+    def parse(fqn: str) -> TargetId:
+        segments = fqn.split('.')
+        if len(segments) != 2:
+            raise ValueError(f'Expected fully qualified target name, got: {fqn!r}')
 
-    res: dict[str, dict[str, Target]] = {}
-    for plugin in plugins:
-        plugin_name = plugin.name
+        plugin, target = segments
 
-        if not _valid_id(plugin_name):
-            _LOGGER.warning(f'Invalid plugin name, skipping: {plugin_name}')
-            continue
+        if not _valid_id(plugin):
+            raise ValueError(f'Invalid plugin identifier: {plugin!r}')
 
-        _LOGGER.info(f'Loading kdist plugin: {plugin_name}')
-        module_name = plugin.value
-        try:
-            _LOGGER.info(f'Importing module: {module_name}')
-            module = importlib.import_module(module_name)
-        except Exception:
-            _LOGGER.error(f'Module {module_name} cannot be imported', exc_info=True)
-            continue
+        if not _valid_id(target):
+            raise ValueError(f'Invalid target identifier: {target!r}')
 
-        res[plugin_name] = _load_targets(module)
+        return TargetId(plugin, target)
 
-    return res
+    @property
+    def fqn(self) -> str:
+        return f'{self.plugin}.{self.target}'
```

Comment:
I assume this is an abbreviation for `fully_qualified_name`? Is it typical to use this abbreviation? If not, maybe we could just use the fully qualified name?

Reply:
I think it's typical, but good point. I renamed it to `full_name`.